pkg/client/client.go

```diff
@@ -863,6 +976,11 @@ func (c *KpmClient) Download(dep *pkg.Dependency, homePath, localPath string) (*
 			return nil, err
 		}
 		dep.FromKclPkg(dpkg)
+		// The downloaded checksum is requested, not calculated
```

Comment:
The download private method is called in many shared methods, but they themselves do not have a file lock added.